src/Microsoft.DotNet.ProjectJsonMigration/MigrationBackupPlan.cs

```diff
@@ -83,8 +85,22 @@ public void PerformBackup()
             {
                 file.MoveTo(
                     Path.Combine(
-                        ProjectBackupDirectory.FullName, file.Name));
+                        ProjectBackupDirectory.FullName,
+                        file.Name));
+            }
+        }
+
+        private static string GetUniqueDirectoryPath(string directoryPath)
+        {
+            var candidatePath = directoryPath;
+
+            var suffix = 1;
+            while (Directory.Exists(directoryPath))
+            {
+                candidatePath = $"{directoryPath}_{suffix++}";
```

Comment:
I think this is fine. The latest migration you ran is the one with the highest number. This is still deterministic. Maybe we should log something in the report?

cc @piotrpMSFT who had concerns about this.